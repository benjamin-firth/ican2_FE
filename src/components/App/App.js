import React, { useEffect } from 'react'
import { Route } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import BeginNavbar from '../BeginNavbar/BeginNavbar';
import Navbar from '../Navbar/Navbar';
import About from '../About/About';
import LoginForm from '../LoginForm/LoginForm';
import SignUpForm from '../SignUpForm/SignUpForm';
import UserProfile from '../UserProfile/UserProfile';
import ProfileContainer from '../ProfileContainer/ProfileContainer';
import Inbox from '../Inbox/Inbox';
import MessageContainer from '../MessageContainer/MessageContainer';
import MeetupContainer from '../MeetupContainer/MeetupContainer';
import { loadMessages } from '../../actions';

const App = () => {
  const dispatch = useDispatch();
  const messages = useSelector(state => state.messages);
  const mentors = useSelector(state => state.mentors);
  const currentUser = useSelector(state => state.currentUser);

  return (
    <main>
      <Route exact path='/'>
        <BeginNavbar />
        <About />
      </Route>
      <Route path='/login'>
        <BeginNavbar />
        <LoginForm />
      </Route>
      <Route path='/signup'>
        <BeginNavbar />
        <SignUpForm />
      </Route>
      <Route path='/myprofile'>
        <Navbar />
        <UserProfile user={useSelector(state => state.currentUser)} />
      </Route>
      <Route exact path='/mentors'>
        <Navbar />
        <ProfileContainer />
      </Route>
      <Route
        exact path='/mentors/:id'
        render={({ match }) => {
          let mentor = mentors.find(mentor => mentor.id === parseInt(match.params.id))
          return mentor &&
            <>
              <Navbar />
              <UserProfile user={mentor} />
            </>
        }}
      />
      <Route path='/inbox'>
        <Navbar />
        <Inbox />
      </Route>
      <Route exact path='/messages'>
        <Navbar />
        <Inbox />
      </Route>
      <Route
        exact path='/messages/:id'
        render={({ match }) => {
          let mentor = mentors.find(mentor => mentor.id === parseInt(match.params.id))
          return mentor &&
            <>
              <Navbar />
              <MessageContainer recipient={mentor} />
            </>
        }}
      />
      <Route path='/meetups'>
        <Navbar />
        <MeetupContainer />
      </Route>
    </main>
  )
}

export default App;
