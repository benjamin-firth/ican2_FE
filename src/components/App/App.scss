@import '../../variables.scss';

* {
  font-family: $primaryfont;
}
