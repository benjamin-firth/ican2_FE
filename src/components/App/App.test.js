import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const mockState = {
  mentors: [{name: 'pants', id: 1}, {name: 'shirts', id: 2}],
  message: [{sender: 2, recipient: 1, body: 'hello'}, {sender: 1, recipient: 2, body: 'hello there'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <App />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
