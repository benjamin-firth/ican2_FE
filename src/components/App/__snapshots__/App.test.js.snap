// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main>
  <Route
    exact={true}
    path="/"
  >
    <BeginNavbar />
    <About />
  </Route>
  <Route
    path="/login"
  >
    <BeginNavbar />
    <LoginForm />
  </Route>
  <Route
    path="/signup"
  >
    <BeginNavbar />
    <SignUpForm />
  </Route>
  <Route
    path="/myprofile"
  >
    <Navbar />
    <UserProfile
      user={
        Object {
          "mentors": Array [
            Object {
              "id": 1,
              "name": "pants",
            },
            Object {
              "id": 2,
              "name": "shirts",
            },
          ],
          "message": Array [
            Object {
              "body": "hello",
              "recipient": 1,
              "sender": 2,
            },
            Object {
              "body": "hello there",
              "recipient": 2,
              "sender": 1,
            },
          ],
        }
      }
    />
  </Route>
  <Route
    exact={true}
    path="/mentors"
  >
    <Navbar />
    <ProfileContainer />
  </Route>
  <Route
    exact={true}
    path="/mentors/:id"
    render={[Function]}
  />
  <Route
    path="/inbox"
  >
    <Navbar />
    <Inbox />
  </Route>
  <Route
    exact={true}
    path="/messages"
  >
    <Navbar />
    <Inbox />
  </Route>
  <Route
    exact={true}
    path="/messages/:id"
  >
    <Navbar />
    <MessageContainer />
  </Route>
  <Route
    path="/meetups"
  >
    <Navbar />
    <MeetupContainer />
  </Route>
</main>
`;
